context/kuberetes_client.go

```diff
@@ -0,0 +1,43 @@
+package context
+
+import (
+	"fmt"
+	"time"
+
+	dutyKubernetes "github.com/flanksource/duty/kubernetes"
+)
+
+type KubernetesClient struct {
+	*dutyKubernetes.Client
+	Connection KubernetesConnection
+	expiry     time.Time
+}
+
+func (c *KubernetesClient) SetExpiry(d time.Duration) {
+	c.expiry = time.Now().Add(d)
+}
+
+func (c *KubernetesClient) RefreshWithExpiry(ctx Context, d time.Duration) error {
```

Comment:
Expiry is something that should come from connection populate, or the JWT token expirty it should. be user specified.